src/StatusQ/Controls/StatusPasswordStrengthIndicator.qml

```diff
@@ -0,0 +1,49 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core 0.1
+
+StatusProgressBar {
+    id: control
+
+    property var strength
+
+    enum Strength {
+        VeryWeak, // 0
+        Weak, // 1
+        SoSo, // 2
+        Good, // 3
+        Great // 4
+    }
+
+    // Behavior:
+    states: [
+        // Strength states definition:
+        State {
+            when: control.strength === StatusPasswordStrengthIndicator.Strength.VeryWeak
+            PropertyChanges { target: control; fillColor : Theme.palette.dangerColor1}
+            PropertyChanges { target: control; text: qsTr("Very weak")}
+        },
+        State {
+            when: control.strength === StatusPasswordStrengthIndicator.Strength.Weak
+            PropertyChanges { target: control; fillColor : Theme.palette.pinColor1}
+            PropertyChanges { target: control; text: qsTr("Weak")}
+        },
+        State {
+            when: control.strength === StatusPasswordStrengthIndicator.Strength.SoSo
+            PropertyChanges { target: control; fillColor : Theme.palette.miscColor7}
+            PropertyChanges { target: control; text: qsTr("So-so")}
+        },
+        State {
+            when: control.strength === StatusPasswordStrengthIndicator.Strength.Good
+            PropertyChanges { target: control; fillColor : Theme.palette.miscColor12}
+            PropertyChanges { target: control; text: qsTr("Good")}
+        },
+        State {
+            when: control.strength === StatusPasswordStrengthIndicator.Strength.Great
+            PropertyChanges { target: control; fillColor : Theme.palette.successColor1}
+            PropertyChanges { target: control; text: qsTr("Great")}
```

Comment:
So far we have avoided putting `qsTr()` calls in StatusQ. The reason being that status desktop will replace all occurances of `qsTr()` with `qsTrId()`.

So essentially, it will change the source of StatusQ when it performs i18n.
Hence, we rather **expose** text properties (with a decent default) so apps like Status Desktop can then assign a `qsTr` from the outside.

Can you please change that accordingly?

Reply:
But by doing that (exposing the text property) you are ‘losing’ some automation in the component itself so you cannot link the variant with the text, only with the colour.  Or do you mean exposing one text property per variant so in this case having 5 different text properties?

And I’m wondering, is there another way of doing that as for example establishing here in StatusQ a unique text ID and then, status-desktop will resolve it accordingly (maybe having a common IDs file or similar)?  For example establishing a hierarchical ID structure like this:
`StatusQ.{ComponentName}.{textId}`

Reply:
> Or do you mean exposing one text property per variant so in this case having 5 different text properties?

Exactly. For now that would be something like:

```
StatusPaswordStrengthIndicator {
  labelVeryWeak: ...
  labelWeak: ...
  labelSoso: ...
  ...
}
```

Again, these properties would have some decent defaults (`"Very weak"`, `"Weak"`, ... etc), but we would give consuming applications full control over how they want to solve i18n.

> And I’m wondering, is there another way of doing that as for example establishing here in StatusQ a unique text ID and then, status-desktop will resolve it accordingly (maybe having a common IDs file or similar)? For example establishing a hierarchical ID structure like this:
StatusQ.{ComponentName}.{textId}

Sounds like a good idea. Feel free to open a separate issue for that where this can be discussed!

Reply:
> Hence, we rather expose text properties (with a decent default) so apps like Status Desktop can then assign a qsTr from the outside.

@PascalPrecht does it mean that if we don't provide values for these exposed properties (by relying on default ones that come with component) the translations will simply not work?

I am almost sure it will result in many bugs related to missing translations. Shouldn't we avoid providing these default strings?

With Qt 5.15 comes the 'required' keyword which ensures each property is assigned by the client who uses the Component.
```
    required property string labelVeryWeak
    required property string labelWeak
    required property string labelSoso
    required property string labelGood
    required property string labelGreat
```

Reply:
> @PascalPrecht does it mean that if we don't provide values for these exposed properties (by relying on default ones that come with component) the translations will simply not work?
 
 If they aren't marked with `qsTr()` the i18n script of Status Desktop won't touch it (otherwise it will replace it with `qsTrId()`.
 
 So yea, it will result in you seeing the default label.
 
 > Shouldn't we avoid providing these default strings?
 
 One could argue that, if no default strings are provided, you'll most likely see many more bugs (e.g. components having no labels at all). So I guess given those two (non-optimal options): which one is worse? No labels and broken UI, or at least english labels in any case?
 
 > With Qt 5.15 comes the 'required' keyword which ensures each property is assigned by the client who uses the Component.
 
 Sounds like a reasonable thing to do then, if we don't come up with a better i18n solution!
 Should be noted that I'm not super familiar with how qsTr/Id() are **supposed** to work. Maybe Desktop isn't doing the right thing in the first place. As in, are apps really required to rewrite their source (from qsTr() -> qsTrId()) to make i18n work? Or is there actually a different, recommended way to use this?

Reply:
> Should be noted that I'm not super familiar with how qsTr/Id() are supposed to work. Maybe Desktop isn't doing the right thing in the first place. As in, are apps really required to rewrite their source (from qsTr() -> qsTrId()) to make i18n work? Or is there actually a different, recommended way to use this?

That's a good point, I've wondered about that before too. In general `qsTr()` is enough for translations, it does not need to be converted to `qsTrId()`. It seems we are doing that to re-use translations from status-react which are id-based: https://hackmd.io/@status-desktop/r1ytYSzn_

I think we could avoid running `qstrConverter.js` (and so the conversion from `qsTr()` to `qsTrId()`) by enhancing the `xmlTranslator.js` script with appropriate logic.